src/domUpdates.js

```diff
@@ -141,19 +138,34 @@ class DomUpdates {
     `)
   }
 
-  updatePurchase(messages, id){
-    let card = document.getElementById(id)
+  updateMessage(messages, id, recipe) {
+    const currentRecipe = recipe.ingredients.map(ing => recipe.ingredientsData.find(recipe => recipe.id === ing.id));
+    let card = document.getElementById(id);
     const values = messages.map( message => message.message)
     card.innerHTML = ``;
-    values.forEach(message => card.innerHTML +=`
-    <ul class="feedback">
-      <li>${message}</li>
-    </ul>
-    `)
+    const newMessage = values.map(message => {
+      const editedMessage = message.split(' ').map(element => {
+        currentRecipe.forEach(data => {
+          if (element === data.id.toString()) {
+            element = data.name;
+          }
+          if (element === '#' || element === 'item') {
+            element = '';
+          }
+        })
+        return element;
+      })
+      return editedMessage.join(' ');
+    });
+    newMessage.forEach(message => card.innerHTML += `
+      <ul class="feedback">
+        <li>${message}</li>
+      </ul>
+      `)
```

Comment:
Looks good, I will try to make this so it doesn't say 'user' or 'unit', and instead says 'you' and the unite for the specific ingredient.

Reply:
It was late when I finished this functionality but removing the user and replacing with `you`, should be ease, let me know if you want me to close the PR when you are done, thanks.